internal/discovery/discovery.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2023 The Kubernetes Authors All rights reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package discovery
+
+import (
+	"time"
+
+	"k8s.io/client-go/discovery"
+	"k8s.io/klog/v2"
+)
+
+// Interval is the interval at which the discovery client fetches the list of objects.
+const Interval = 30 * time.Second
```

Comment:
It is not very useful to have the default interval exposed in the discovery package as it is only used in the server code. We could directly move it there.

Reply:
Using this in the CRS package now in addition to the server code, to error out the endless [`for` loop](https://github.com/kubernetes/kube-state-metrics/pull/1851/files#r1093460954)  if the condition isn't met in time (GVKs are not filled before timeout).